thunderfish/consistentfishes.py

```diff
@@ -42,67 +45,67 @@ def find_consistency(fundamentals, df_th = 1.):
     :return consistent_fundamentals: (1-D array) List containing all values that are available in all given lists.
     :return index:          (1-D array) Indices of the values that are in every list relating to the fist list in fishlists.
     """
-    consistency_help = np.ones_like(fundamentals[0])
-
-    index = []
-    consistent_fundamentals = []
+    consistency_help = np.ones(len(fundamentals[0]), dtype=int)
 
     for enu, fundamental in enumerate(fundamentals[0]):
-        for list in np.arange(len(fundamentals)-1)+1:
-            freq_diff = [fundamentals[list][i] - fundamental for i in np.arange(len(fundamentals[list]))]
-            for i in np.arange(len(freq_diff)):
-                if freq_diff[i] >= -df_th and freq_diff[i] <= df_th:
-                    consistency_help[enu] += 1
-                    break
-
-    for idx in np.arange(len(consistency_help)):
-        if consistency_help[idx] == len(fundamentals):
-            index.append(idx)
-            consistent_fundamentals.append(fundamentals[0][idx])
+        for list in range(1, len(fundamentals)):
+            if np.sum(np.abs(fundamentals[list] - fundamental) < df_th) > 0:
+                consistency_help[enu] += 1
+
+    index = np.arange(len(fundamentals[0]))[consistency_help == len(fundamentals)]
```

Comment:
No. numpy.nonzero gives back the indices where the array is not zero. Here we are looking for the indices where value is the same as the number of given fishlists.